templates/todo/projects/nodejs-mongo/.repo/terraform/infra/output.tf

```diff
@@ -16,7 +16,7 @@ output "REACT_APP_WEB_BASE_URL" {
 }
 
 output "REACT_APP_API_BASE_URL" {
-  value = module.api.URI
+  value = var.useAPIM ? module.apimApi[0].SERVICE_API_URI : module.api.URI

```

Comment:
is this magic number `0` resistant to changes in the future where there might be more index?

Reply:
This is a result of how to do conditional resource deployments in terraform. It's not meant to be a n-th length array, rather a 1-length array.

Terraform by default requires all resources specified to be non-optional (and required). To make a resource optional, you need to declare the resource with a `count`, with the value `0` when the resource isn't expected to be deployed, otherwise the value `1`:

`count                    = var.useAPIM ? 1 : 0`

Thus, this is simply indexing into the only element expected.